src/Orleans.TestingHost/AppDomainSiloHost.cs

```diff
@@ -79,15 +78,8 @@ public AppDomainTestHooks(ISiloHost host)
             this.host = host;
         }
 
-        internal IBootstrapProvider GetBootstrapProvider(string name)
-        {
-            var bootstrapProviderManager = this.host.Services.GetRequiredService<BootstrapProviderManager>();
-            IBootstrapProvider provider = (IBootstrapProvider)bootstrapProviderManager.GetProvider(name);
-            return CheckReturnBoundaryReference("bootstrap provider", provider);
-        }
-
         /// <summary>Find the named storage provider loaded in this silo. </summary>
-        internal IStorageProvider GetStorageProvider(string name) => CheckReturnBoundaryReference("storage provider", (IStorageProvider)this.host.Services.GetRequiredService<StorageProviderManager>().GetProvider(name));
+        internal IStorageProvider GetStorageProvider(string name) => CheckReturnBoundaryReference("storage provider", this.host.Services.GetServiceByName<IStorageProvider>(name));
```

Comment:
throw `KeyNotFOundException` to preserve original logic, which was maintained by `StorageProviderManager`